app/views/smart_proxies/show.html.erb

```diff
@@ -1,73 +1,25 @@
+
   <ul id="proxy-tab" class="nav nav-tabs">
-    <li class="active"><a href="#properties" data-toggle="tab"><%= _('Overview') %></a></li>
-    <% if service_features.any? %>
-      <li><a href="#services" data-toggle="tab"><%= _('Services') %></a></li>
+    <% @page.sorted_tabs.each_with_index do |(name, tab), index| %>
+      <% next if name == "services" && service_features.none? %>
+      <li class="<%= 'active' if index == 0 %>">
+        <a href="#<%= tab.snake_name %>" data-toggle="tab"><%= _(name.capitalize) %></a>
+      </li>
     <% end %>
   </ul>
+
   <div id="proxy-tab-content" class="proxy-content tab-content">
-    <div class="tab-pane active in" id="properties">
-      <div class="col-md-6">
-        <div class="row">
-          <h3><%= _('Details') %></h3>
-        </div>
-        <div class="row">
-          <div class="col-md-4">
-            <%= _('Communication status') %>
-          </div>
-          <div class="col-md-8">
-            <span class="proxy-show-status">
-              <%= spinner %>
-            </span>
-          </div>
-        </div>
-        <div class="row">
-          <div class="col-md-4">
-            <%= _('URL') %>
-          </div>
-          <div class="col-md-8">
-            <%= @smart_proxy.url %>
-          </div>
-        </div>
-        <div class="row">
-          <div class="col-md-4">
-            <%= _('Version') %>
-          </div>
-          <div class="col-md-8">
-            <span class="proxy-version" data-url="<%= ping_smart_proxy_path(@smart_proxy) %>">
-              <%= spinner %>
-            </span>
-          </div>
-        </div>
-        <div class="row">
-          <div class="col-md-4">
-            <%= _('Features') %>
-          </div>
-          <div class="col-md-8">
-            <%= @smart_proxy.features.to_sentence %> <%= refresh_proxy_icon(@smart_proxy, authorizer) %>
-          </div>
-        </div>
-      </div>
-      <div class="col-md-6">
-        <!--place holder for Capsule sync-->
-      </div>
-    </div>
-    <% if service_features.any? %>
-      <div class="tab-pane" id="services">
-        <div class="col-md-6">
-          <% service_features.each do |feature| %>
-            <% feature_erb = "smart_proxies/plugins/#{feature}" %>
-            <% if lookup_context.template_exists?(feature_erb, [], true) %>
-              <%= render :partial => feature_erb, :locals => {:feature => feature} %>
-            <% else %>
-              <%= render :partial => "smart_proxies/plugins/no_template", :locals => {:feature => feature} %>
-            <% end %>
-          <% end %>
-        </div>
+    <% @page.sorted_tabs.each_with_index do |(name, tab), index| %>
+      <% next if name == 'services' && service_features.none? %>
+      <div class="tab-pane <%= 'active' if index == 0 %>" id="<%= tab.snake_name %>">
+        <%= render tab.layout, { :tab => tab, :smart_proxy => @smart_proxy, :service_features => service_features } %>
       </div>
     <% end %>
```

Comment:
These blocks could be helper fragments or methods, so developers can simply create a container and use them.